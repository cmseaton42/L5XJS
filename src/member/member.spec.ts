import { Member } from './index';

describe('Member Class', () => {
  describe('New Instance', () => {
    it('Accepts Proper Input', () => {
      const fn = (options: any) => () => new Member(options);

      expect(fn({ datatype: 'DINT' })).toThrow();
      expect(fn({ name: null, datatype: 'DINT' })).toThrow();
      expect(fn({ name: 'tagName', datatype: 'DINT' })).not.toThrow();
      expect(fn({ name: 'tagName', datatype: 'DINT', description: 12 })).toThrow();
      expect(fn({ name: 'tagName', datatype: 'DINT', description: 'A Description' })).not.toThrow();
      expect(fn({ name: 'tagName', datatype: 'DINT', description: 'A Description', hidden: 12 })).toThrow();
      expect(fn({ name: 'tagName', datatype: 'DINT', description: 'A Description', hidden: true })).not.toThrow();
      expect(fn({ name: 'tagName', datatype: 'DINT', description: 'A Description', hidden: true, target: 12 })).toThrow();
      expect(fn({ name: 'tagName', datatype: 'DINT', description: 'A Description', hidden: true, target: 'aTarget' })).not.toThrow();
      expect(fn({ name: 'tagName', datatype: 'DINT', description: 'A Description', hidden: true, target: 'aTarget', bitNumber: 'notANumber' })).toThrow();
      expect(fn({ name: 'tagName', datatype: 'DINT', description: 'A Description', hidden: true, bitNumber: 12 })).toThrow();
      expect(fn({ name: 'tagName', datatype: 'DINT', description: 'A Description', hidden: true, target: 'aTarget', bitNumber: 12 })).not.toThrow();
    });

    it('Initializes with Desired Document Model', () => {
      let mem = new Member({ name: 'tagName', datatype: 'DINT' });
      expect(mem).toMatchSnapshot();

      mem = new Member({ name: 'tagName', datatype: 'BOOL', target: 'aTarget', bitNumber: 0 });
      expect(mem).toMatchSnapshot();

      mem = new Member({ name: 'tagName', datatype: 'BIT', target: 'aTarget', bitNumber: 0 });
      expect(mem).toMatchSnapshot();

      mem = new Member({ name: 'tagName', datatype: 'DINT', description: 'A Description' });
      expect(mem).toMatchSnapshot();

      mem = new Member({ name: 'tagName', datatype: 'DINT', description: 'A Description', hidden: true });
      expect(mem).toMatchSnapshot();

      mem = new Member({ name: 'tagName', datatype: 'DINT', description: 'A Description', hidden: false, target: 'ATarget' });
      expect(mem).toMatchSnapshot();

      mem = new Member({ name: 'tagName', datatype: 'DINT', description: 'A Description', hidden: false, target: 'ATarget', bitNumber: 12 });
      expect(mem).toMatchSnapshot();
    });
  });

  describe('Static Methods', () => {
    test('isMember: Returns Appropriate Judgement', () => {
      let mem = new Member({ name: 'tagName', datatype: 'DINT' });
      expect(Member.isMember(mem)).toBeTruthy();

      const notMember = { notATag: 12 };
      expect(Member.isMember(notMember)).toBeFalsy();

      expect(Member.isMember(12)).toBeFalsy();
    });
  });
});
