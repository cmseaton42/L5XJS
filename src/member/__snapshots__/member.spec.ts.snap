// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member Class New Instance Initializes with Desired Document Model 1`] = `
Member {
  "_dom": {
    "attributes": {
      "DataType": "DINT",
      "Hidden": false,
      "Name": "tagName",
      "Radix": "decimal",
    },
    "elements": [],
    "name": "Member",
    "type": "element",
  },
}
`;

exports[`Member Class New Instance Initializes with Desired Document Model 2`] = `
Member {
  "_dom": {
    "attributes": {
      "BitNumber": 0,
      "DataType": "BIT",
      "Hidden": false,
      "Name": "tagName",
      "Radix": "binary",
      "Target": "aTarget",
    },
    "elements": [],
    "name": "Member",
    "type": "element",
  },
}
`;

exports[`Member Class New Instance Initializes with Desired Document Model 3`] = `
Member {
  "_dom": {
    "attributes": {
      "BitNumber": 0,
      "DataType": "BIT",
      "Hidden": false,
      "Name": "tagName",
      "Radix": "binary",
      "Target": "aTarget",
    },
    "elements": [],
    "name": "Member",
    "type": "element",
  },
}
`;

exports[`Member Class New Instance Initializes with Desired Document Model 4`] = `
Member {
  "_dom": {
    "attributes": {
      "DataType": "DINT",
      "Hidden": false,
      "Name": "tagName",
      "Radix": "decimal",
    },
    "elements": [
      {
        "elements": [
          {
            "text": "A Description",
            "type": "text",
          },
        ],
        "name": "Description",
        "type": "element",
      },
    ],
    "name": "Member",
    "type": "element",
  },
}
`;

exports[`Member Class New Instance Initializes with Desired Document Model 5`] = `
Member {
  "_dom": {
    "attributes": {
      "DataType": "DINT",
      "Hidden": true,
      "Name": "tagName",
      "Radix": "decimal",
    },
    "elements": [
      {
        "elements": [
          {
            "text": "A Description",
            "type": "text",
          },
        ],
        "name": "Description",
        "type": "element",
      },
    ],
    "name": "Member",
    "type": "element",
  },
}
`;

exports[`Member Class New Instance Initializes with Desired Document Model 6`] = `
Member {
  "_dom": {
    "attributes": {
      "DataType": "DINT",
      "Hidden": false,
      "Name": "tagName",
      "Radix": "decimal",
    },
    "elements": [
      {
        "elements": [
          {
            "text": "A Description",
            "type": "text",
          },
        ],
        "name": "Description",
        "type": "element",
      },
    ],
    "name": "Member",
    "type": "element",
  },
}
`;

exports[`Member Class New Instance Initializes with Desired Document Model 7`] = `
Member {
  "_dom": {
    "attributes": {
      "DataType": "DINT",
      "Hidden": false,
      "Name": "tagName",
      "Radix": "decimal",
    },
    "elements": [
      {
        "elements": [
          {
            "text": "A Description",
            "type": "text",
          },
        ],
        "name": "Description",
        "type": "element",
      },
    ],
    "name": "Member",
    "type": "element",
  },
}
`;
